pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -102,7 +102,15 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 	server := kapp.NewKubeletServer()
 	server.Config = path
 	server.RootDirectory = options.VolumeDirectory
-	server.HostnameOverride = options.NodeName
+
+	// kubelet finds the node IP address by doing net.ParseIP(hostname) and if that fails,
+	// it does net.LookupIP(NodeName) and picks the first non-loopback address.
+	// Pass node IP as hostname to make kubelet use the desired IP address.
+	if len(options.NodeIP) > 0 {
+		server.HostnameOverride = options.NodeIP
```

Comment:
I'm confused about what this is doing.  The field helps says that this is for pod traffic, but I thought that `--hostname-override` on the node controlled how the master communicated with the node and affected the default nodeName if none was specified (usually cloud provider I think).